modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala

```diff
@@ -112,8 +112,16 @@ object BlockConsensusCell {
   private def deriveConsensusPeerIds(proposal: Proposal, selfId: PeerId): Set[PeerId] =
     proposal.facilitators + proposal.senderId + proposal.owner - selfId
 
-  private def returnTransactions[F[_]](ownProposal: Proposal, transactionStorage: TransactionStorage[F]): F[Unit] =
-    transactionStorage.put(ownProposal.transactions)
```

Comment:
Maybe we should keep them as hashed transactions, but unwrap them only when sending to facilitators?